modules/charon-core/src/main/java/org/wso2/charon3/core/attributes/DefaultAttributeFactory.java

```diff
@@ -90,17 +90,16 @@ public static Attribute createAttribute(AttributeSchema attributeSchema,
     }
 
     /**
-     * When an attribute is created with value and data type provided,
+     * When an attribute is created with value and data type provided,.
      * we need to validate whether they are matching.
      *
      * @param attributeValue
      * @param attributeDataType
      * @return boolean
      * @throws BadRequestException
      */
-    protected static boolean isAttributeDataTypeValid(Object attributeValue,
-                                                      SCIMDefinitions.DataType attributeDataType)
-                                                throws BadRequestException {
+    protected static boolean isAttributeDataTypeValid(Object attributeValue, SCIMDefinitions.DataType attributeDataType)
```

Comment:
We are not recommending to change public methods as we need to preserve the backward compatibility as these public methods can be used from different other places.